### PR TITLE
SNOW-823356: Local testing add tests to union functions

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/mock/mock_plan.py [hunk 1]
```diff
@@ -528,18 +537,27 @@ def calculate_expression(
...
+    *,
+    keep_literal: bool = False,
+) -> Union[ColumnEmulator, object]:
     """Returns the calculated expression evaluated based on input table/column"""
     if isinstance(exp, Attribute):
-        return input_data[expr_to_alias.get(exp.expr_id, exp.name)]
+        try:
+            return input_data[expr_to_alias.get(exp.expr_id, exp.name)]
+        except KeyError:
+            # expr_id maps to the projected name, but input_data might still have the exp.name
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/mock/mock_plan.py:548]: What's an example that hits this case?

File: src/snowflake/snowpark/mock/mock_plan.py [hunk 2]
```diff
@@ -528,18 +537,27 @@ def calculate_expression(
     input_data: Union[TableEmulator, ColumnEmulator],
     analyzer,
     expr_to_alias: Dict[str, str],
-) -> ColumnEmulator:
+    *,
+    keep_literal: bool = False,
+) -> Union[ColumnEmulator, object]:
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/mock/mock_plan.py:542]: Does the `object` here correspond to when `keep_literal` is True?

Comment (sfc-gh-aling) [reply src/snowflake/snowpark/mock/mock_plan.py]: comment updated

File: src/snowflake/snowpark/mock/mock_plan.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/mock/mock_plan.py:552]: added comment for context + todo

**Pull request merged**